Snippets/mongodb/MongoDB_3/DocumentVersion.cs

```diff
@@ -16,7 +16,7 @@ async Task UpdateWithVersion(IMongoCollection<BsonDocument> collection, string v
             //Define other update operations on the document
 
             var modifiedDocument = await collection.FindOneAndUpdateAsync<BsonDocument>(
-                filter: document => document["_id"] == documentId && document["_version"] == currentVersion,
+                filter: document => document["_id"].AsGuid == documentId && document["_version"] == currentVersion,
```

Comment:
Is this working? See https://github.com/Particular/NServiceBus.Storage.MongoDB/issues/108 
I might be confusing things, though. Just double-checking

Reply:
might depend on the specific configuration and the client being used tbh. This is just a sample, alternatives would be to:
* replace `documentId` with `BsonBinaryData(guid, GuidRepresentation.X)` (where X also depends)
* change the sample to not use a GUID as the id
thoughts?

Reply:
I struggle to answer this because I'm lacking client knowledge. Maybe we don't have to overthink it and just leave it as is